source/extensions/filters/http/aws_lambda/config.cc

```diff
@@ -32,6 +32,19 @@ getInvocationMode(const envoy::extensions::filters::http::aws_lambda::v3::Config
   PANIC_DUE_TO_CORRUPT_ENUM;
 }
 
+Extensions::Common::Aws::CredentialsProviderSharedPtr
+getCredentialsProvider(const std::string& profile,
+                       Server::Configuration::ServerFactoryContext& server_context,
+                       const std::string& region) {
+  if (!profile.empty()) {
+    return std::make_shared<Extensions::Common::Aws::CredentialsFileCredentialsProvider>(
+        server_context.api(), profile);
+  }
+  return std::make_shared<Extensions::Common::Aws::DefaultCredentialsProviderChain>(
+      server_context.api(), makeOptRef(server_context), region,
+      Extensions::Common::Aws::Utility::fetchMetadata);
+}
```

Comment:
Due to this new config only has meaning for AWS Credentials File provider, in case it is set, only that provider will be taken in consideration to avoid getting wrong credentials. If a user set this config is because it expect to get the credentials from the file and should failed if the file is not present or the env var does is missed or incorrectly set.

Reply:
Can you add code comments for the function and around if condition?

Reply:
Also a debug log will be useful to clearly state that profile is provided in the lambda config. So only credentials file provider will be used. Other credentials provider will be ignored.